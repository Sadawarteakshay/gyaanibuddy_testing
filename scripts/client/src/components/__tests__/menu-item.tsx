import React from 'react';

import { render, cleanup } from '../../test-utils';
import MenuItem from '../menu-item';
import { renderApollo } from '../../test-utils';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('Menu Item', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    renderApollo(<MenuItem to="/wow" />);
  });

  it('render snapshot', () => {
    const PageContainerElement = <MenuItem to="/wow" />;

    // see https://stackoverflow.com/questions/70805929/how-to-fix-error-usehref-may-be-used-only-in-the-context-of-a-router-compon
    render(PageContainerElement, {wrapper: MemoryRouter});
    
    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ElementTree = renderer
      .create(<MemoryRouter>{PageContainerElement}</MemoryRouter>)
      .toJSON();

    expect(ElementTree).toMatchSnapshot();
  });
});
