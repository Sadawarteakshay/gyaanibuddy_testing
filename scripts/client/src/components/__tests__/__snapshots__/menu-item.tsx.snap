// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item render snapshot 1`] = `
<a
  className="css-j1pn2v"
  href="/wow"
  onClick={[Function]}
/>
`;
